homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,261 @@
+"""
+Support for GitHub.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.github/
+"""
+from datetime import timedelta
+import logging
+import json
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN, CONF_USERNAME, CONF_PASSWORD, CONF_PATH, CONF_NAME,
+    CONF_SCAN_INTERVAL)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['PyGithub==1.43.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_REPOS = 'repositories'
+
+ATTR_PATH = 'path'
+ATTR_NAME = 'name'
+ATTR_STARGAZERS = 'stargazers'
+ATTR_TOPICS = 'topics'
+ATTR_BRANCHES = 'branches'
+ATTR_OPEN_ISSUES = 'open_issues'
+ATTR_OPEN_PULL_REQUESTS = 'open_pull_requests'
+ATTR_LAST_COMMIT = 'last_commit'
+
+DEFAULT_NAME = 'GitHub'
+
+SCAN_INTERVAL = timedelta(seconds=300)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_REPOS): cv.ensure_list
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the GitHub sensor platform."""
+    interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    sensors = []
+    for repository in config[CONF_REPOS]:
+        if CONF_ACCESS_TOKEN in config:
+            data = GitHubData(
+                interval=interval,
+                repository=repository,
+                access_token=config[CONF_ACCESS_TOKEN]
```

Comment:
For optional config options use `config.get(key)` this will return `None` if the key does not exist.

Reply:
:+1: